tests/common.go

```diff
@@ -3886,3 +3887,110 @@ func IsCloudDriveInitialised(n node.Node) (bool, error) {
 	}
 	return false, err
 }
+
+//WaitForExpansionToStart waits for pool expansion to trigger
+func WaitForExpansionToStart(poolID string) error {
+	f := func() (interface{}, bool, error) {
+		expandedPool, err := GetStoragePoolByUUID(poolID)
+
+		if err != nil {
+			return nil, false, err
+		}
+		if expandedPool.LastOperation != nil {
+			if expandedPool.LastOperation.Status == opsapi.SdkStoragePool_OPERATION_FAILED {
+				return nil, false, fmt.Errorf("PoolResize has failed. Error: %s", expandedPool.LastOperation)
+			}
+
+			if expandedPool.LastOperation.Status == opsapi.SdkStoragePool_OPERATION_IN_PROGRESS {
+				// storage pool resize has been triggered
+				logrus.Infof("Pool %s expansion started", poolID)
+				return nil, true, nil
+			}
+		}
+		return nil, true, fmt.Errorf("pool %s resize not triggered ", poolID)
+	}
+
+	_, err := task.DoRetryWithTimeout(f, 2*time.Minute, 10*time.Second)
+	return err
+}
+
+//RebootNodeAndWait reboots node and waits for to be up
+func RebootNodeAndWait(n node.Node) error {
+
+	if &n == nil {
+		return fmt.Errorf("no Node is provided to reboot")
+	}
+
+	err := Inst().N.RebootNode(n, node.RebootNodeOpts{
+		Force: true,
+		ConnectionOpts: node.ConnectionOpts{
+			Timeout:         1 * time.Minute,
+			TimeBeforeRetry: 5 * time.Second,
+		},
+	})
+
+	if err != nil {
+		return err
+	}
+	err = Inst().N.TestConnection(n, node.ConnectionOpts{
+		Timeout:         15 * time.Minute,
+		TimeBeforeRetry: 10 * time.Second,
+	})
+	if err != nil {
+		return err
+	}
+	err = Inst().V.WaitDriverDownOnNode(n)
+	if err != nil {
+		return err
+	}
+	err = Inst().S.IsNodeReady(n)
+	if err != nil {
+		return err
+	}
+	err = Inst().V.WaitDriverUpOnNode(n, Inst().DriverStartTimeout)
+	if err != nil {
+		return err
+	}
+
+	return nil
+
+}
+
+//GetNodeWithGivenPoolID returns node having pool id
```

Comment:
Is there any reason this method is not part of node driver?

Reply:
there is a dependency on GetStorageNodes function which inturn dependent on volume driver.
moving to node driver is giving dependency errors